docs/design.md

```diff
@@ -34,7 +34,20 @@ Bevy is still young, and many design patterns are still being discovered and ref
 Most do not even have an agreed name yet. For some prior work in this area that inspired us,
 see [the Unofficial Bevy Cheatbook](https://bevy-cheatbook.github.io/) and [bevy_best_practices](https://github.com/tbillington/bevy_best_practices).
 
-## Code structure
+## Pattern Table of Contents
+
+- [Plugin Organization](#plugin-organization)
+- [Widgets](#widgets)
+- [Asset Preloading](#asset-preloading)
+- [Spawn Command](#spawn-command)
+- [Interaction Callback](#interaction-callback)
+- [Dev Tools](#dev-tools)
+- [Screen State](#screen-state)
```

Comment:
```suggestion
- [Spawn Commands](#spawn-commands)
- [Interaction Callbacks](#interaction-callbacks)
- [Dev Tools](#dev-tools)
- [Screen States](#screen-states)
```
Subjective but I _think_ these read better pluralized.

Reply:
"The spawn command pattern" - "The spawn commands pattern"
"The interaction callback pattern" - "The interaction callbacks pattern"
"The screen state pattern" - "The screen states pattern"

Hmm, I'd go for singular to make it easier to have a common language.

Reply:
"The widget pattern" or "The widgets pattern" then?

Reply:
"The widget pattern" sounds way more natural to me. Good point: [`e3d3ebb` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/e3d3ebb2fdab8cd0273eb18def9f35670734afd1)

Reply:
I like the plural versions better personally. Also there's still "The dev tools pattern".

Reply:
Can we just have the headers as the plural versions and allow people to use plural or singular w/e they want when they append "pattern" at the end? :P

Reply:
Alright: [`ff40632` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/ff40632f7cb0adf1defa28116ea7ec546a084e2d)